### PR TITLE
ACI Support and AKS Terraform

### PR DESCRIPTION
**Pull request by tedmartn**
ACI and AKS both working using Terraform.

### CODE REVIEW

File: deploy-aci/main.tf [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+### Defines Azure Resource Manager Terraform Provider ###
+#########################################################
+provider "azurerm" {
+    version = "2.9.0"
+    features {}
+}
+#########################################################
+######### Creates Azure Resource Group ##################
+#########################################################
+resource "azurerm_resource_group" "useast-minecraftaci-rg" {
```
Comment (x0techdad) [inline deploy-aci/main.tf]: @tedmartn  please update RG name so it lines up with RG name pattern on existing AKS deployment "rg-cooldad-mvm"

Comment (x0techdad) [reply deploy-aci/main.tf]: also, if the location is brought in via a parameter and can change, dont hardcode the location name in this resource name

File: deploy-aci/main.tf [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  name = var.resource_group_name
+  location = var.resource_location
+}
+
+#########################################################
+######### Creates Azure Storage Account #################
+#########################################################
+
+resource "azurerm_storage_account" "useast-aci-storage" {
```
Comment (x0techdad) [inline deploy-aci/main.tf]: if the location is brought in via a parameter and can change, dont hardcode the location name in this resource name

File: deploy-aci/main.tf [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  name = "useastacistorage"
```
Comment (x0techdad) [inline deploy-aci/main.tf]: can resource names be created by allowing a custom prefix variable with some static content, something like this:

`name = format("sta",var.resource_name_prefix,"001")`

File: deploy-aci/main.tf [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  account_tier = "Premium"
+  account_replication_type = "LRS"
+  account_kind = "FileStorage"
+
+}
+
+#########################################################
+######### Creates 100GB Azure File Share#################
+#########################################################
+resource "azurerm_storage_share" "useast-aci-share" {
```
Comment (x0techdad) [inline deploy-aci/main.tf]: if the location is brought in via a parameter and can change, dont hardcode the location name in this resource name

File: deploy-aci/main.tf [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  name = "acishare"
```
Comment (x0techdad) [inline deploy-aci/main.tf]: shared name should reflect use case, maybe something like this:

` name = "pvc-mc-001"`

File: deploy-aci/main.tf [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  storage_account_name = azurerm_storage_account.useast-aci-storage.name
```
Comment (x0techdad) [inline deploy-aci/main.tf]: if the location is brought in via a parameter and can change, dont hardcode the location name in this resource name

File: deploy-aci/main.tf [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  quota = 100
+}
+
+#########################################################
+######### Creates Azure Container Instance Group ########
+########## 1 Minecraft Container is Created #############
+########### 1 CPU and 2 GB Memory Allocated #############
+#########################################################
+resource "azurerm_container_group" "useast-minecraftaci-cg" {
```
Comment (x0techdad) [inline deploy-aci/main.tf]: if the location is brought in via a parameter and can change, dont hardcode the location name in this resource name

File: deploy-aci/main.tf [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  name = "useast-minecraft-aci"
+  location = azurerm_resource_group.useast-minecraftaci-rg.location
+  resource_group_name = azurerm_resource_group.useast-minecraftaci-rg.name
```
Comment (x0techdad) [inline deploy-aci/main.tf]: please update RG name so it lines up with RG name pattern on existing AKS deployment "rg-cooldad-mvm"

File: deploy-aci/main.tf [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  quota = 100
+}
+
+#########################################################
+######### Creates Azure Container Instance Group ########
+########## 1 Minecraft Container is Created #############
+########### 1 CPU and 2 GB Memory Allocated #############
+#########################################################
+resource "azurerm_container_group" "useast-minecraftaci-cg" {
+  name = "useast-minecraft-aci"
```
Comment (x0techdad) [inline deploy-aci/main.tf]: can resource names be created by allowing a custom prefix variable with some static content, something like this:

`name = format("aci-",var.resource_name_prefix,"-001")`

File: deploy-aci/main.tf [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+  location = azurerm_resource_group.useast-minecraftaci-rg.location
+  resource_group_name = azurerm_resource_group.useast-minecraftaci-rg.name
+  ip_address_type = "public"
+  dns_name_label = "tedstestaci"
```
Comment (x0techdad) [inline deploy-aci/main.tf]: make unique by using gui, let me know if you need example

**Review by x0techdad — CHANGES_REQUESTED** [review]
Please address the comments in-line. Reach out with any Qs

**Comment by tedmartn** [comment]
Check out the most recent commit, I believe I have addressed all of your concerns bro.

File: deploy-aks/variables.tf [hunk 11]
```diff
@@ -1,15 +1,6 @@
-variable "resource_group_name" {
-  description = "Please input resource group name"
-}
 variable "resource_location" {
   description = "Please input resource location"
 }
-variable "useast-minecraft-aksname" {
```
Comment (x0techdad) [inline deploy-aks/variables.tf]: dont hardcode location identifier into code if users are allowed to select their preferred region.

**Review by x0techdad — CHANGES_REQUESTED** [review]
One  code change.  Once completed, code review will be done and we can move on to testing.

File: deploy-aks/.terraform.lock.hcl [hunk 12]
```diff
@@ -0,0 +1,42 @@
+# This file is maintained automatically by "terraform init".
```
Comment (x0techdad) [inline deploy-aks/.terraform.lock.hcl]: is this file required?
seems like thats local to your client and probably dont want to include it in the project?

File: deploy-aks/main.tf [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    virtual_network_name = azurerm_virtual_network.mvm-aks-vnet.name
+}
+
+
+#########################################################
+######### Creates Azure Kubernetes Cluster ##############
+######### 1 Minecraft Container is Created ##############
+########## 2 CPU and 8 GB Memory Allocated ##############
+#########################################################
+resource "azurerm_kubernetes_cluster" "useast-minecraft-aks" {
```
Comment (x0techdad) [inline deploy-aks/main.tf]: Do not hard code location, allow user to pick deployment location or inherit from resource group location.

File: deploy-aks/main.tf [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+    admin_username = var.admin_username
+
+    ssh_key {
+      key_data = file(var.ssh_public_key)
+    }
+    }
+   default_node_pool {
+     name = "default"
+     node_count = 2
+     vm_size = "standard_d2s_v4"
```
Comment (x0techdad) [inline deploy-aks/main.tf]: set to burstable size.

**Review by x0techdad — CHANGES_REQUESTED** [review]

File: deploy-aks/main.tf (hunk 13 above)
Comment (tedmartn) [reply deploy-aks/main.tf]: Definitely changed this, are you looking at the same commit as me?

File: deploy-aks/main.tf (hunk 14 above)
Comment (tedmartn) [reply deploy-aks/main.tf]: Done.

File: deploy-aks/.terraform.lock.hcl (hunk 12 above)
Comment (tedmartn) [reply deploy-aks/.terraform.lock.hcl]: Excluded with gitignore file
